x/profiles/keeper/keeper_chain_links.go

```diff
@@ -95,3 +94,64 @@ func (k Keeper) DeleteAllUserChainLinks(ctx sdk.Context, user string) {
 		k.DeleteChainLink(ctx, link)
 	}
 }
+
+// getOldestUserChainByChain returns the oldest chain link of the given owner associated to the given chain name
+// If the such the chain link is not exists, return false instead.
+func (k Keeper) getOldestUserChainByChain(ctx sdk.Context, owner, chainName string) (types.ChainLink, bool) {
+	var oldestLink types.ChainLink
+	found := false
+	k.IterateUserChainLinksByChain(ctx, owner, chainName, func(link types.ChainLink) (stop bool) {
+		if !found {
+			oldestLink = link
+			found = true
+			return false
+		}
+
+		if link.CreationTime.Before(oldestLink.CreationTime) {
+			oldestLink = link
+		}
+		return false
+	})
+
+	return oldestLink, found
+}
+
+// updateOwnerDefaultExternalAddress updates the default external address of the given chain name for the given owner
+// It must be performed after deleting the default external address chain link
```

Comment:
Why is it necessary to do this?

Reply:
It updates the default address to the oldest chain link if the user unlink the default one. The idea is that the default address must exist if the profile has chain links.